### ISSUE TITLE
unable to update package

### ISSUE THREAD

**Issue by theZulqarnain** [feren-OS/feren-OS-Bug-Reporting#53]
when i'm updating feren-maintenacne from update managers i'm getting this error.
**E: feren-ppas: subprocess installed post-installation script returned error exit**

**Comment by TITUXFEREN** [comment]
Hi, I already have the same problem: I can not install or remove software because I retrieve the same error message : **E: No se pudo bloquear /var/cache/apt/archives/lock - open (11: Resource temporarily unavailable)**
 everything was going on great until I upgraded the system... now I can not install software, please help me!!!

**Comment by feren** [comment]
@TITUXFEREN That means that APT is current running in the background, and is currently doing a package operation. That's completely normal. Just wait a little while, and eventually it'll go past that message when ran.

**Comment by theZulqarnain** [comment]
for me it showing every thing is ok but feren Maintenance not updating

**Comment by feren** [comment]
If it only says that Mintsources failed, and feren-maintenance was in the updates list, then Feren Maintenance should've updated successfully

**Comment by feren** [comment]
@theZulqarnain @TITUXFEREN @monk10101
Pushed an update to 'feren-maintenance' that means it can update without feren-ppas blocking it, so that once rebooted, the OS can fix itself...

**Comment by theZulqarnain** [comment]
two times i rebooted, but same problems

**Comment by feren** [comment]
...once this is done, hopefully after either rebooting your machine, or launching Feren Maintenance manually, the OS can fix the add PPA command, fixing the installation failures...

**Comment by feren** [comment]
...you'll have to update 'feren-maintenance' again first, then reboot

**Comment by feren** [comment]
...or does it get blocked by apt telling you to run an apt-get -f install first?

**Comment by theZulqarnain** [comment]
when i'm updating after reboot i'm getting **E: feren-ppas: subprocess installed post-installation script returned error exit status 1** and for apt-get -f install it shows me **E: Could not get lock /var/cache/apt/archives/lock - open (11: Resource temporarily unavailable)
E: Unable to lock directory /var/cache/apt/archives/**

**Comment by feren** [comment]
Wait a little bit, that just means APT is checking for any updates, as normal...
About to update the package anyway to make the fix more forceful to APT...

**Comment by feren** [comment]
@theZulqarnain Alright, could you pop up a Terminal, and run the following please?
```sudo apt update
sudo apt install feren-maintenance -y
sudo feren-maintenance```

**Comment by feren** [comment]
@theZulqarnain Alright, could you pop up a Terminal, and run the following please?

```
sudo apt update
sudo apt install feren-maintenance -y
sudo feren-maintenance
```

**Comment by feren** [comment]
sudo password = your password

**Comment by feren** [comment]
...don't worry about any errors relating to feren-ppas during the second command, as the third command should then fix it.

Once the final command is done, then try `sudo apt-get -f install` again.

**Comment by feren** [comment]
@theZulqarnain Is it working so far?

**Comment by monk10101** [comment]
ran update. all looks good now.   Thank You! :)

**Comment by theZulqarnain** [comment]
for me same error

**Comment by feren** [comment]
@monk10101 Is it completely fixed now? apt-get -f install successful now?

**Comment by feren** [comment]
@theZulqarnain Did you run `sudo apt-get -f install` after all the commands in that comment above, to check?

**Comment by theZulqarnain** [comment]
yeah it show me **E: Could not open lock file /var/lib/dpkg/lock - open (2: No such file or directory)
E: Unable to lock the administration directory (/var/lib/dpkg/), are you root?**  , i think some thing went wrong with my os

**Comment by feren** [comment]
Did you forget to run the commands with 'sudo'? That's saying you aren't running the commands as Root, and 'sudo' runs them as root.

**Comment by theZulqarnain** [comment]
no ran with sudo

**Comment by feren** [comment]
Try 'sudo -i' in the Terminal, followed by running the commands

**Comment by monk10101** [comment]
i read :

@theZulqarnain @TITUXFEREN @monk10101
Pushed an update to 'feren-maintenance' that means it can update without feren-ppas blocking it, so that once rebooted, the OS can fix itself...

so I ran update (it completed with out errors) then rebooted checked again for updates (none).
appears to be fixed . i never ran apt-get -f install

**Comment by zaivala** [comment]
zaivala@zaivala-ThinkPad-T430 ~ $ sudo apt install feren-maintenance -y
Reading package lists... Done
Building dependency tree       
Reading state information... Done
feren-maintenance is already the newest version (2018.11.0.0.8).
0 upgraded, 0 newly installed, 0 to remove and 3 not upgraded.
2 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Setting up feren-ppas (2018.05.0.0.1) ...
deb [trusted=yes] http://sourceforge.net/projects/feren-os-repositories/files/stable/ ./
OK
Traceback (most recent call last):
  File "/usr/lib/linuxmint/mintSources/mintSources.py", line 1593, in <module>
    codename = config_parser.get("general", "base_codename")
  File "/usr/lib/python2.7/ConfigParser.py", line 330, in get
    raise NoSectionError(section)
ConfigParser.NoSectionError: No section: 'general'
dpkg: error processing package feren-ppas (--configure):
 subprocess installed post-installation script returned error exit status 1
dpkg: dependency problems prevent configuration of feren-system-packages:
 feren-system-packages depends on feren-ppas; however:
  Package feren-ppas is not configured yet.

dpkg: error processing package feren-system-packages (--configure):
 dependency problems - leaving unconfigured
No apport report written because the error message indicates its a followup error from a previous failure.
                          Errors were encountered while processing:
 feren-ppas
 feren-system-packages
E: Sub-process /usr/bin/dpkg returned an error code (1)
zaivala@zaivala-ThinkPad-T430 ~ $ 

Something wrong in this step...

**Comment by theZulqarnain** [comment]
what the hell, after running all those commands my whole user folders removed. unable to enter any user folders like download ,pictures etc

**Comment by feren** [comment]
Huh? That for sure isn't what the commands above are designed to do in Feren OS...
Open Nemo As Root (Right-click on ANY FOLDER, even Desktop (if desktop icons are enabled), and hit 'Open as Root'), and check if the folders are accessible under there, if they are, open up a Terminal, and run the following:
```
USERNM=$(whoami)
sudo chown -hR $USERNM:$USERNM /home/$USERNM
```

This should give you back your Permissions to your Home Folder and folders inside of it, if that's why you can't access them all of a sudden.

**Comment by feren** [comment]
Anyway, here's the final fix to this bug:
1. Go into Update Manager
2. Refresh
3. Select only to update 'feren-maintenance' to make sure we don't get any other errors other than 'feren-ppas' and packages that depend on it
4. Update 'feren-maintenance'
5. Run Feren Maintenance from the menu, or reboot (it launches on every bootup), and let it finish (it's rather quick)
6. Now update the rest of the packages pending an update, and all should be fixed now.

PS: If there's still that Home issue @theZulqarnain that all I can think is an error in command running, because none of the scripts nor commands ran even go as far as touch the /home directory, or anything inside there, then start a new issue here about it.
